File: src/main/java/codingdojo/StoreEvent.java
package codingdojo;

/**
 * Ticket to In-store event, eg wine tasting or book signing
 */
public class StoreEvent extends Item {
    private Store sellingStore;
    private Store location;

    public StoreEvent(String name, Store location) {
        super(name, "EVENT", 0);
        this.location = location;
    }

    public void setBillingData(Store store) {
        this.sellingStore = store;
    }


    public void setLocation(Store locationStore) {
        this.location = locationStore;
    }
}

File: typescript/src/StoreEvent.ts
import Item from "./Item";
import Store from "./Store";

/**
 * Ticket to In-store event, eg makeover, eye shadow master class
 * or beauty product launch evening reception.
 */
export default class StoreEvent extends Item {
    protected location!: Store;

    constructor(name: string, location: Store) {
        super(name, "EVENT", 0);
        this.setLocation(location);
    }

    public setLocation(locationStore: Store) {
        this.location = locationStore;
        this.location.addStoreEvent(this);
    }

    public toString() {
        return "StoreEvent{" +
            "name='" + this.name + '\'' +
            ", location=" + this.location +
            '}';
    }
}
